uploader/templates/upload.html

```diff
@@ -1,383 +1,408 @@
+  }
+  // A simple callback implementation.
+  function pickerCallback(data) {
+    if (data.action == google.picker.Action.PICKED) {
+      data.docs.forEach(item =>
+        pickedFiles.push({
+          name: item.name,
+          id: item.id,
+          description: item.description
+        })
+      );
+
+      $(".step-3").show();
+      $(".step-1-2").hide();
+
+      fileStagingTable = new Tabulator("#picked-files-table", {
+        data: pickedFiles,
+        layout: "fitColumns",
+        columns: [
+          {
+            title: "File",
+            field: "id",
+            formatter: function(cell, formatterParams, onRendered) {
+              return `<img src='${generateThumbNailLink(cell.getValue())}'>`;
+            },
+            width: 80
+          },
+          {
+            title: "Title",
+            field: "name",
+            editor: "input",
+            validator: "unique"
+          },
+          {
+            title: "Description",
+            field: "description",
+            editor: "textarea",
+            validator: "required"
+          }
+        ]
+      });
+      fileStagingTable.setData(pickedFiles);
+    }
+  }
+
+  function uploadFiles() {
+    // Disable the button first.
+    $("#upload-button").toggleClass("disabled");
+    var fileData = {
+      fileList: fileStagingTable.getData(),
+      token: oauthToken
+    };
+
+    function getCookie(name) {
+      var cookieValue = null;
+      if (document.cookie && document.cookie !== "") {
+        var cookies = document.cookie.split(";");
+        for (var i = 0; i < cookies.length; i++) {
+          var cookie = cookies[i].trim();
+          // Does this cookie string begin with the name we want?
+          if (cookie.substring(0, name.length + 1) === name + "=") {
+            cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+            break;
+          }
+        }
+      }
+      return cookieValue;
+    }
+
+    var csrftoken = getCookie("csrftoken");
+
+    function csrfSafeMethod(method) {
+      // these HTTP methods do not require CSRF protection
+      return /^(GET|HEAD|OPTIONS|TRACE)$/.test(method);
+    }
+
+    $.ajax({
+      // prettier-ignore
+      url: "{{ request.scheme }}://{{ request.META.HTTP_HOST }}{% url 'file_upload' %}",
+      type: "POST",
+      data: JSON.stringify(fileData),
+      cache: false,
+      processData: false,
+      contentType: "application/json",
+      beforeSend: function(xhr, settings) {
+        if (!csrfSafeMethod(settings.type) && !this.crossDomain) {
+          xhr.setRequestHeader("X-CSRFToken", csrftoken);
+        }
+      },
+      success: function(data) {
+        $(".step-3").hide();
+        $(".step-1-2").hide();
+        $(".step-4").show();
+        $(".results-panel").append("<tbody>");
+        data.forEach(function(item) {
+          $(".results-panel").append(
+            "<tr><td><img src='" +
+              generateThumbNailLink(item.url) +

```

Comment:
Sadly, we cannot use this at this point. item.url is a link to the wiki file. We should simply use the same link and probably add some width and height stylings to it.